src/components/Pensjonssimulering/utils.ts

```diff
@@ -174,6 +227,93 @@ export function handleChartScroll(event: Event) {
   }
 }
 
+export const getHoverColor = (previousColor: string): string => {
+  switch (previousColor) {
+    case SERIE_COLOR_INNTEKT: {
+      return SERIE_COLOR_FADED_INNTEKT
+    }
+    case SERIE_COLOR_AFP: {
+      return SERIE_COLOR_FADED_AFP
+    }
+    case SERIE_COLOR_TP: {
+      return SERIE_COLOR_FADED_TP
+    }
+    case SERIE_COLOR_ALDERSPENSJON: {
+      return SERIE_COLOR_FADED_ALDERSPENSJON
+    }
+    default: {
+      return ''
+    }
+  }
+}
+
+export const getNormalColor = (previousColor: string): string => {
+  switch (previousColor) {
+    case SERIE_COLOR_FADED_INNTEKT: {
+      return SERIE_COLOR_INNTEKT
+    }
+    case SERIE_COLOR_FADED_AFP: {
+      return SERIE_COLOR_AFP
+    }
+    case SERIE_COLOR_FADED_TP: {
+      return SERIE_COLOR_TP
+    }
+    case SERIE_COLOR_FADED_ALDERSPENSJON: {
+      return SERIE_COLOR_ALDERSPENSJON
+    }
+    default: {
+      return previousColor
+    }
+  }
+}
+
+export function onPointClick(this: Point, event: PointClickEventObject): void {
+  const pointIndex = event.point.index
+  this.series.chart.series.forEach(function (serie: Series) {
+    serie.data.forEach(function (point: Point) {
+      const color =
+        point.index !== pointIndex
+          ? getHoverColor(point.color as string)
+          : getNormalColor(point.color as string)
+
+      if (color && color !== point.color) {
+        point.update({ color }, false)
+      }
+    })
+  })
+  ;(
+    this.series.chart.xAxis[0] as ExtendedAxis
+  ).labelGroup.element.childNodes.forEach(function (
+    label: HTMLElement,
+    index: number
+  ) {
+    if (index === pointIndex) {
+      label.style.fontWeight = 'bold'
+    } else {
+      label.style.fontWeight = 'normal'
+    }
+  })
+  this.series.chart.redraw()
+}
+
+export function onChartClick(this: Chart): void {
```

Comment:
Kanskje denne kan ha et navn som beskriver hva den gjør?

Reply:
Denne funksjonen er en event handler. Det er grunnen til at jeg velger å forholde meg til konvensjonen
`on*` som blir da til `'onChartClick'`. I lik linje som `'onPointClick'` osv.  Den kan evt. hete `'handleChartClick'` for å være mer nøyaktig, men det blir fort ulesbart om vi begynner å beskrive selve handlingen i navnet.

Beskrivelse av implementasjonsdetaljene er dokumentert konsekvent gjennom enhetstestene. Så her finner man mer beskrivelse:
https://github.com/navikt/pensjonskalkulator-frontend/pull/123/files#diff-7194dc19a8c045d67d99ef9a6b4d17b982107c6b3e9165f5ed9a92a2d03a35d8R407

Reply:
Kode blir tvert imot mer lesbar ved at man gir funksjoner beskrivende navn. Et unntak er kanskje hvis funksjonen gjør så mange forskjellige ting at det blir vanskelig å gi et kort og passe beskrivende navn. I det tilfellet her kunne kanskje `updateColorsAndLabels` eller noe i den duren. Det er mye lettere å umiddelbart forstå hva som skjer her:
```javascript
events: {
  click: updateColorsAndLabels
}
```
enn her:
```javascript
events: {
  click: onChartClick
}
```
Bare ment som et tips, er ikke fremmed for at vi beholder navnet du allerede har.

Reply:
Takk for tips! Jeg tok dette alternativet i vurdering og opplever det ikke nødvendigvis som mer lesbart, men heller unøyaktig.
Jeg konkluderte at jeg foretrekker å holde meg til en etablert konvensjon og heller ha en korrekt beskrivelse og dokumentasjon gjennom enhetstestene.